### PR TITLE
feat: add checkpoint 

### PR DESCRIPTION
**Pull request by mohandast52**
- checkpoint call added

Screenshoeeee

### CODE REVIEW

File: components/Home/Donate/index.jsx [hunk 1]
```diff
@@ -180,6 +204,15 @@ export const DepositServiceDonation = () => {
...
           </EpochStatus>
         ))}
+
+        <Button
+          type="primary"
+          loading={isCheckpointLoading}
+          disabled={isExpectedEndTimeInFuture}
+          onClick={onCheckpoint}
+        >
+          Checkpoint
```
Comment (oaksprout) [inline components/Home/Donate/index.jsx:214]:
```suggestion
          Start new epoch
```

Comment (oaksprout) [reply components/Home/Donate/index.jsx:214]: Internally maybe it makes sense to refer to checkpoint, but it's not explicit for an external user what this means.

File: components/Home/Donate/requests.jsx [hunk 2]
```diff
@@ -81,3 +81,16 @@ export const getVeOlasThresholdRequest = async () => {
...
+
+export const checkpointRequest = async (account) => {
+  const contract = getTokenomicsContract();
+  try {
+    const fn = contract.methods.checkpoint().send({ from: account });
+
+    const response = await sendTransaction(fn, account);
+    return response?.transactionHash;
+  } catch (error) {
+    notifyError('Error occured on checkpoint');
```
Comment (oaksprout) [inline components/Home/Donate/requests.jsx:93]:
```suggestion
    notifyError('Error – could not start new epoch');
```

File: components/Home/Donate/index.jsx [hunk 3]
```diff
@@ -126,6 +131,25 @@ export const DepositServiceDonation = () => {
...
 
+  const onCheckpoint = async () => {
+    try {
+      setIsCheckpointLoading(true);
+
+      await checkpointRequest(account);
+
+      // update epoch details after checkpoint
+      const { updatedEpochCounter } = await getThresholdData();
+      notifySuccess(`New epoch started ${updatedEpochCounter}`);
```
[the remark this answers is not in the document]
Comment (oaksprout) [reply components/Home/Donate/index.jsx:142]: Yes good call to avoid a page refresh.

Comment (oaksprout) [inline components/Home/Donate/index.jsx:142]:
```suggestion
      notifySuccess(`Started new epoch`);
```

File: components/Home/Donate/index.jsx [hunk 4]
```diff
@@ -180,6 +204,15 @@ export const DepositServiceDonation = () => {
...
         ))}
+
+        <Button
+          type="primary"
+          loading={isCheckpointLoading}
+          disabled={isExpectedEndTimeInFuture}
+          onClick={onCheckpoint}
+        >
+          Checkpoint
+        </Button>
```
Comment (oaksprout) [inline components/Home/Donate/index.jsx:215]: We don't have a good docs setup on this site like Contribute, so I'd suggest a line of help text here:

`<Text type="secondary">New epochs must be manually triggered by community members</Text>`

**Review by kupermind — APPROVED** [review]

**Comment by kupermind** [comment]
Please check the linter

**Review by oaksprout — CHANGES_REQUESTED** [review]

**Review by oaksprout — APPROVED** [review]

**Pull request merged**